From c170c41062518a5f5f0c4c16feda6cdc07010c75 Mon Sep 17 00:00:00 2001
From: Ileana Dumitrescu <ileanadumitrescu95@gmail.com>
Date: Wed, 11 Dec 2024 18:43:47 +0200
Subject: [PATCH] libtoolize.in: Create symlinks with mklink for MSVC

Symlink files are not generated properly with 'ln -s' for MSVC to
process, so mklink is used, which swaps the argument order of target
and link_name. This also requires absolute paths, generated with the
use of cygpath.

* libtoolize.in: Check if linking with mklink and update processing.

---
diff --git a/libtoolize.in b/libtoolize.in
index 77d0c9ed..b576ef58 100644
--- a/libtoolize.in
+++ b/libtoolize.in
@@ -437,6 +437,31 @@ func_copy ()
     else
       my_copycmd=$LN_S
       my_copy_type=linking
+
+      # The Windows linker, mklink, swaps link_name and target.
+      case $my_copycmd in
+      *mklink*)
+        my_temp=$my_destfile
+        my_destfile=$my_srcfile
+        my_srcfile=$my_temp
+
+        my_copy_msg="$my_copy_type file '$my_destfile'"
+        $opt_verbose && my_copy_msg="$my_copycmd $my_srcfile $my_destdir"
+
+        my_destfile=`cygpath -aw "$my_destfile"`
+        my_srcfile=`cygpath -aw "$my_srcfile"`
+
+        if test -d "$my_destfile"; then
+          case $host_os in
+          mingw*) # MSYS
+            my_copycmd="$my_copycmd //d" ;;
+          *)
+            my_copycmd="$my_copycmd /d" ;;
+          esac
+        fi
+      ;;
+      *) ;;
+      esac
     fi
     my_copy_msg="$my_copy_type file '$my_destfile'"
     $opt_verbose && my_copy_msg="$my_copycmd $my_srcfile $my_destdir"
--
2.45.2

